### PR TITLE
Fix: defeat http.Client memory leak

### PR DESCRIPTION
**Pull request by AlexStocks**
u should check t.Client is nil or not again in mutex lock

### CODE REVIEW

File: pkg/cloudevents/transport/http/transport.go [hunk 1]
```diff
@@ -170,7 +170,9 @@ func (t *Transport) Send(ctx context.Context, event cloudevents.Event) (context.
 func (t *Transport) obsSend(ctx context.Context, event cloudevents.Event) (context.Context, *cloudevents.Event, error) {
 	if t.Client == nil {
 		t.crMu.Lock()
-		t.Client = &http.Client{}
+		if t.Client == nil {
```
Comment (n3wscott) [inline pkg/cloudevents/transport/http/transport.go]: this code can only get here if `t.Client == nil`, see line 171

Comment (AlexStocks) [reply pkg/cloudevents/transport/http/transport.go]: however, this is not thread safe. It there are two goroutines exec this func concurrently, the first create the t.Client at line 173, and the other is blocked at line 172. Later on the first goroutine release the lock  at line 174, and the second goroutine will get the lock and create the t.Client again at line 173.

Comment (n3wscott) [reply pkg/cloudevents/transport/http/transport.go]: ah I see, can you kill line 171?

Comment (n3wscott) [reply pkg/cloudevents/transport/http/transport.go]: well... nevermind, no need to toggle that lock over and over....

**Comment by n3wscott** [comment]
LGTM

**Pull request merged**